src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -0,0 +1,30 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import TreeTransforms._
+import Contexts.Context
+import Decorators._
+import pickling._
+import config.Printers
+
+/** This miniphase pickles trees */
+class Pickler extends MiniPhaseTransform { thisTransform =>
```

Comment:
It's not a miniphase. I would say it's better to make this explicit by making this a `MacroTransform`